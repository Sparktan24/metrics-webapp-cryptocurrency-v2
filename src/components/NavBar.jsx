/* eslint-disable import/no-extraneous-dependencies */
import { NavLink } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { resetCoins } from '../redux/coins/coinsSlice';
import backArrow from '../styles/back_arrow.png';

const NavBar = () => {
  const dispatch = useDispatch();
  return (
    <nav className="nav-bar">
      <div className="nav-div">
        <NavLink to="/HomePage" onClick={() => dispatch(resetCoins())}>
          <img src={backArrow} className="backArrow" alt="back arrow" />
        </NavLink>
        {/* <NavLink to="/CoinsDetail">CoinDetails</NavLink> */}
      </div>
      <span className="nav-bar-title">Top Cryptocurrencies</span>
    </nav>
  );
};

export default NavBar;
