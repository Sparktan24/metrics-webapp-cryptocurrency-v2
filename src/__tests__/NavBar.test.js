import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import NavBar from '../components/NavBar';
import store from '../app/store';

describe('NavBar', () => {
  it('render correctly', () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <NavBar />
        </Router>
      </Provider>,
    );
    expect(screen.getByText('Top Cryptocurrencies')).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders back arrow', () => {
    render(
      <Provider store={store}>
        <Router>
          <NavBar />
        </Router>
      </Provider>,
    );
    expect(screen.getByAltText('back arrow')).toBeInTheDocument();
  });
});
