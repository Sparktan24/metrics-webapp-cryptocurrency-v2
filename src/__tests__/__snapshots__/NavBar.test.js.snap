// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar render correctly 1`] = `
<nav
  class="nav-bar"
>
  <div
    class="nav-div"
  >
    <a
      class=""
      href="/HomePage"
    >
      <img
        alt="back arrow"
        class="backArrow"
        src="back_arrow.png"
      />
    </a>
  </div>
  <span
    class="nav-bar-title"
  >
    Top Cryptocurrencies
  </span>
</nav>
`;
